standalone-dns-proxy/cmd/standalonednsproxy.go

```diff
@@ -75,6 +77,30 @@ func (sdp *StandaloneDNSProxy) StartStandaloneDNSProxy() error {
 	return nil
 }
 
+// watchConnection watches the connection state
+func (sdp *StandaloneDNSProxy) WatchConnection(ctx context.Context, _ cell.Health) error {
```

Comment:
Do you need this job? What about a separate WithTrigger job that actually starts the proxy? Then, when the first set of rules is received in `handlePolicyStream()`, trigger that? Or, have the proxy be a job and it waits for a channel to be closed.

But these are minor suggestions, non-blocking.